peekaboo/ruleset/rules.py

```diff
@@ -248,10 +248,9 @@ async def evaluate(self, sample):
         """ Try to get information about the sample from the database. Return
         the old result and reason if found and advise the engine to stop
         processing. """
-        ktreport = await self.get_knowntools_report(sample)
-        if ktreport.known:
-            result, reason = ktreport.worst()
-            return self.result(result, reason, False)
+        worst = await self.db_con.analysis_journal_get_worst(sample)
```

Comment:
In my opinion the Known Rule and the Known Tools Report are two distinct features, with a similar functionality and name.
They are in different parts of the ruleset.conf and there is no need for one to depend on the other.

Reply:
Agreed. But what's your opinion regarding the trade-off on multiple ruleset roundtrips? Should we cache the known rule result in the sample as well to avoid repeared database requests? Or just stay with the knowntools analyser usage for that reason?

Reply:
I would think the difference in marginal (which makes the decicion more difficult).
Caching won't keep the result between delivery attempts, so only if we can actually wait and poll for an analyser to finish.
Hence when we abort the ruleset with the `PeekabooAnalysisDeferred` exception which is only used by the explicit `CuckooRule` if I'm correct.

Reply:
The cuckooreport and cortexreport expression identifiers do it as well.

Reply:
This has gained a new aspect: Since we're discussing moving the duplicate checks out of the sample submit and closer to long running analysers we should keep the known rule check as light as possible in terms of database load.

Reply:
> This has gained a new aspect: Since we're discussing moving the duplicate checks out of the sample submit and closer to long running analysers we should keep the known rule check as light as possible in terms of database load.

This has just been decided for us: Working on that very feature I was just bitten by the fact that a cached known rule result totally subverts duplicate detection: If we delay samples in some Cuckoo or the expressions rule and later resubmit them to the queue, they'll have been through the ruleset once already and have a cached knowntools report that says that they're not known to the system. Otherwise they wouldn't have made it to the rule that delayed them as duplicates in the first place. :/ So we need the known rule to directly access the database so it picks up results of concurrent analyses of the same sample.